src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrPusher.cs

```diff
@@ -0,0 +1,216 @@
+        var body = @"{""query"" : ""{" + query + @"}""}";
+
+        HttpResponseMessage? response = null;
+        var url = "https://api.github.com/graphql";
+
+        _httpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue("Darc", null));
+
+        using HttpRequestMessage message = new(HttpMethod.Post, url);
+        message.Content = new StringContent(body, Encoding.UTF8, "application/json");
+        message.Headers.Authorization = new AuthenticationHeaderValue("Token", gitHubApiPat);
+
+        try
+        {
+            response = await _httpClient.SendAsync(message, cancellationToken);
+            response.EnsureSuccessStatusCode();
+        }
+        catch (HttpRequestException)
+        {
+            _logger.LogError("Cannot connect to GraphQl api");
+            throw;
+        }
+
+        var content = response.Content.ReadAsStringAsync(cancellationToken).Result;
+        var settings = new JsonSerializerOptions
+        {
+            PropertyNameCaseInsensitive = true,
+        };
+
+        var result = JsonSerializer.Deserialize<CommitsQueryResult>(content, settings);
+        return result;
+    }
+
+    private CommitSearchArguments GetCommitSearchArguments(string uri, string sha)
+    {
+        var gitHubUrlPrefix = "https://github.com";
+        var azDoUrlPrefix = "https://dev.azure.com";
+
+        if (uri.StartsWith(azDoUrlPrefix, StringComparison.OrdinalIgnoreCase))
+        {
+            string[] repoParts = uri.Substring(uri.LastIndexOf('/')).Split('-', 2);
+
+            if (repoParts.Length != 2)
+            {
+                throw new Exception($"Invalid URI in source manifest. Repo '{uri}' does not end with the expected <GH organization>-<GH repo> format.");
+            }
+
+            string org = repoParts[0];
+            string repo = repoParts[1];
+
+            // The internal Nuget.Client repo has suffix which needs to be accounted for.
+            const string trustedSuffix = "-Trusted";
+            if (uri.EndsWith(trustedSuffix, StringComparison.OrdinalIgnoreCase))
+            {
+                repo = repo.Substring(0, repo.Length - trustedSuffix.Length);
+            }
+
+            return new CommitSearchArguments(repo, org, sha);
+        }
+        else if (uri.StartsWith(gitHubUrlPrefix, StringComparison.OrdinalIgnoreCase))
+        {
+            string[] repoParts = uri.Substring(gitHubUrlPrefix.Length).Split('/', StringSplitOptions.RemoveEmptyEntries);
+
+            if (repoParts.Length != 2)
+            {
+                throw new Exception($"Invalid URI in source manifest. Repo '{uri}' does not end with the expected <GH organization>/<GH repo> format.");
+            }
+
+            return new CommitSearchArguments(repoParts[1], repoParts[0], sha);
+        }
+
+        throw new Exception("Unsupported repo url format.");
+    }
+
+    private string SanitizeName(string repoName) => repoName.Replace("-", string.Empty).Replace(".", string.Empty);
```

Comment:
Can we name this `GetGraphQlIdentifier` or similar to make it clear why and how we sanitize?